lib/complex.ex

```diff
@@ -964,22 +1167,198 @@ defmodule Complex do
+  end
+
+  @doc """
+  Calculates erf(x) of the argument
+  """
+
+  if math_fun_supported?(:erf, 1) do
+    def erf(x) when is_number(x) do
+      :math.erf(x)
+    end
+  else
+    def erf(x) when is_number(x) do
+      x = x |> max(-4.0) |> min(4.0)
+      x2 = x * x
+
+      alpha =
+        0.0
+        |> muladd(x2, -2.72614225801306e-10)
+        |> muladd(x2, 2.77068142495902e-08)
+        |> muladd(x2, -2.10102402082508e-06)
+        |> muladd(x2, -5.69250639462346e-05)
+        |> muladd(x2, -7.34990630326855e-04)
+        |> muladd(x2, -2.95459980854025e-03)
+        |> muladd(x2, -1.60960333262415e-02)
+
+      beta =
+        0.0
+        |> muladd(x2, -1.45660718464996e-05)
+        |> muladd(x2, -2.13374055278905e-04)
+        |> muladd(x2, -1.68282697438203e-03)
+        |> muladd(x2, -7.37332916720468e-03)
+        |> muladd(x2, -1.42647390514189e-02)
+
+      min(x * alpha / beta, 1.0)
+    end
   end
+
+  def erf(%Complex{re: re, im: im}) do
+    if im != 0 do
+      raise ArithmeticError, "erf not implemented for non-real numbers"
+    end
+
+    Complex.new(erf(re))
+  end
+
+  if math_fun_supported?(:erfc, 1) do
+    def erfc(z) when is_number(z), do: :math.erfc(z)
+  end
+
+  def erfc(z), do: subtract(1, erf(z))
+
+  def erf_inv(z) when is_number(z) do
+    w = -:math.log((1 - z) * (1 + z))
+    erf_inv_p(w) * z
+  end
+
+  def erf_inv(%Complex{re: re, im: im}) do
+    if im != 0 do
+      raise ArithmeticError, "erf_inv not implemented for non-real numbers"
+    end
+
+    Complex.new(erf_inv(re))
+  end
+
+  defp erf_inv_p(w) when w < 5 do
+    w = w - 2.5
+
+    2.81022636e-08
+    |> muladd(w, 3.43273939e-07)
+    |> muladd(w, -3.5233877e-06)
+    |> muladd(w, -4.39150654e-06)
+    |> muladd(w, 0.00021858087)
+    |> muladd(w, -0.00125372503)
+    |> muladd(w, -0.00417768164)
+    |> muladd(w, 0.246640727)
+    |> muladd(w, 1.50140941)
+  end
+
+  defp erf_inv_p(w) do
+    w = :math.sqrt(w) - 3
+
+    -0.000200214257
+    |> muladd(w, 0.000100950558)
+    |> muladd(w, 0.00134934322)
+    |> muladd(w, -0.00367342844)
+    |> muladd(w, 0.00573950773)
+    |> muladd(w, -0.0076224613)
+    |> muladd(w, 0.00943887047)
+    |> muladd(w, 1.00167406)
+    |> muladd(w, 2.83297682)
+  end
+
+  defp muladd(acc, t, n) do
+    acc * t + n
+  end
+
+  ### Logical Operations
+  @doc "Logical equality between two numbers"
+  def equal?(a, b), do: a == b
+
+  @doc "Logical difference between two numbers"
+  def not_equal?(a, b), do: a != b
+
+  @doc """
+  Logical AND between two numbers.
+
+  Zero is false, everything else is true.
+  """
+  def logical_and?(a, b), do: as_boolean(a) and as_boolean(b)
+
+  @doc """
+  Logical OR between two numbers.
+
+  Zero is false, everything else is true.
+  """
+  def logical_or?(a, b), do: as_boolean(a) or as_boolean(b)
+
+  @doc """
+  Logical XOR between two numbers.
+
+  Zero is false, everything else is true.
+  """
+  def logical_xor?(a, b), do: as_boolean(a) != as_boolean(b)
```

Comment:
I am feeling those logical ones do not really belong here. Maybe let's keep them exclusively in Nx for now?

Reply:
Sure. I'll open a PR there and remove them from here